testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/shared/SharedSessionFailoverTestCase.java

```diff
@@ -92,14 +91,16 @@
 
     @Test
     public void test(
-            @ArquillianResource(SimpleServlet.class) @OperateOnDeployment(DEPLOYMENT_1) URL baseURL1,
-            @ArquillianResource(SimpleServlet.class) @OperateOnDeployment(DEPLOYMENT_2) URL baseURL2)
+            @ArquillianResource @OperateOnDeployment(DEPLOYMENT_1) URL baseURLDep1,
+            @ArquillianResource @OperateOnDeployment(DEPLOYMENT_2) URL baseURLDep2)
             throws URISyntaxException, IOException {
+        URL baseURL1 = new URL(baseURLDep1.toExternalForm() + "/");
```

Comment:
@pferraro thank you for review, I have reworked the changes to use more URI based approach and to limit URL -> String -> URL conversions. Done in separate commit. If it looks good to you, I shall squash the commits.